app/nginx-1.11.10/src/event/ngx_event_connect.c

```diff
@@ -38,7 +38,23 @@ ngx_event_connect_peer(ngx_peer_connection_t *pc)
 
     type = (pc->type ? pc->type : SOCK_STREAM);
 
+#if (NGX_HAVE_FSTACK)
+    /* 
+     We need to explicitly call the needed socket() function
+         to create socket!
+    */
+    static int (*real_socket)(int, int, int);
+    if (pc->belong_to_host) {
+        if (!real_socket) {
+            real_socket = dlsym(RTLD_NEXT, "socket");
+        }
+        s = real_socket(pc->sockaddr->sa_family, type, 0);
+    } else {
+        s = ngx_socket(pc->sockaddr->sa_family, type, 0);
+    }
```

Comment:
Is there a graceful way? I know that it is a little difficult here.

Reply:
Maybe we can add a creation flag ```SOCK_CREKERNEL```(create-by-kernel), so that this part can be implemented in the glue layer‘s socket().
```
#define SOCK_CREKERNEL 0x1000
```
Here, or when we want to create socket by kernel, we code like this : 
```
s = ngx_socket(pc->sockaddr->sa_family, type | SOCK_CREKERNEL , 0);
```
Also, ```fstack_territory``` needs to be changed.

What do you think?

Reply:
It's a better way.
What about defining `SOCK_FSTACK`(create a f-stack socket)? Thus we need to modify sockets that are created by f-stack.
But it's up to you.